tripal_chado/includes/TripalImporter/OBOImporter.inc

```diff
@@ -1063,27 +1068,63 @@ class OBOImporter extends TripalImporter {
             '!url' => $full_url,
           ]), TRIPAL_WARNING);
       }
-      //What should happen with this stuff?
-      $base_iri = $ontology_results['config']['baseUris'][0];
-      $ontologyID = $ontology_results['ontologyId'];
-      $this->baseIRIs[$short_name] = [$ontologyID, $base_iri];
+      // The following foreach code works but, I am not sure that
+      // I am retrieving each query result from the json associative
+      // array with the correct style
+      foreach ($ontology_results['response']['docs'] as $each ) {
+        $obo_id = $each['obo_id'];
+        $defining_ontology = $each['is_defining_ontology'];
+        // First result should have defining_ontology=true, but if 
+        // it doesn't, use the first result with obo_id=$id
+	if ($defining_ontology == 'false' and $obo_id != $id) {
+	  continue;
+	}
+        $iri = urlencode(urlencode($each['iri']));
+        $ontologyID = $each['ontology_name'];
+	// Type should be 'property' or 'class' in the response
+        $type = $each['type'];
+      }
     }
 
     // Next get the term.
-    $iri = urlencode(urlencode($base_iri . $accession));
-    $full_url = 'http://www.ebi.ac.uk/ols/api/ontologies/' . $ontologyID . '/terms/' . $iri;
-    $response = drupal_http_request($full_url, []);
-    if (!$response) {
-      throw new Exception(t('Did not get a response from EBI OLS trying to lookup term: !id',
-        ['!id' => $id]));
-    }
-    $results = drupal_json_decode($response->data);
+    $query = $type;
+    if ($type == 'class') {
+      $query = 'term';
+    }
+    $results =  $this->oboEbiLookup($id, $query);
     if (!$results) {
-      $this->logMessage('Error: no data with !url.  The response was: !response', [
-        '!url' => $full_url,
-        '!response' => $response,
-      ]);
-
+      $query = 'query-non-local';
+      $ontology_results = $this->oboEbiLookup($id, $query);
+      if ($ontology_results) {
+        foreach ($ontology_results['response']['docs'] as $each ){
+          $obo_id = $each['obo_id'];
+          $defining_ontology = $each['is_defining_ontology'];
+          if ( !$defining_ontology and $obo_id != $id ) {    
+            continue;
+          }  
+      	  $found_iri = urlencode(urlencode($each['iri']));
+          $ontology = $each['ontology_name'];
+	  // Type should be 'property' or 'class' in the response
+          $type = $each['type'];
+	  if ($type == 'property') {
+            $type = 'properties';
+	  }
+	  elseif ( $type == 'class') {
+            $type = 'terms';
+          }
+	  $results = $this->oboEbiLookup($id, $type, $found_iri);
+        } 
+      }
```

Comment:
More indention issues.

Reply:
Weird, it "looks" fine in my editor. i will delete the all the white space and try again.